### PR TITLE
Fix the error in Category selector

### PR DESCRIPTION
**Pull request by soyeongmin**

### CODE REVIEW

File: src/Components/CategoryInput.tsx [hunk 1]
```diff
@@ -11,20 +11,26 @@ interface Props {
...
-  function removeElement(arr: string[], item: string) {
-    return arr.filter(e => e != item)
+  function selectElement(item: string) {
+    if (selections.includes(item)) {
+      setSelections(selections.filter(s => s != item))
+    } else setSelections([...selections, item])
+    getCategory(selections)
+  }
+
+  function deleteElement(item: string) {
```
Comment (jhw123) [inline src/Components/CategoryInput.tsx:31]: it is better to use `useCallback` hook when defining functions in a component!

```typescript
const deleteElement = useCallback((item: string) => {
    setCategories(categories.filter(c => c != item))
    setSelections(selections.filter(s => s != item))
}, [setSelections, selections.filter])
```

File: src/Components/CategoryInput.tsx [hunk 2]
```diff
@@ -11,20 +11,26 @@ interface Props {
...
+    else if (inputCat == '') alert('Please enter the category') //no input case
     else alert('You have the same category') //duplicates case
   }
 
-  function removeElement(arr: string[], item: string) {
-    return arr.filter(e => e != item)
+  function selectElement(item: string) {
+    if (selections.includes(item)) {
+      setSelections(selections.filter(s => s != item))
+    } else setSelections([...selections, item])
```
Comment (jhw123) [inline src/Components/CategoryInput.tsx:27]:
```suggestion
    } else {
      setSelections([...selections, item])
    }
```

Comment (jhw123) [reply src/Components/CategoryInput.tsx:27]: personal preference though

File: src/Components/CategoryInput.tsx [hunk 3]
```diff
@@ -11,20 +11,26 @@ interface Props {
 }
 
 export const CategoryInput = ({ getCategory }: Props) => {
-  const inputRef = useRef<string>('')
-  //Should get category data from db
+  const [inputCat, setInputCat] = useState<string>('')
   const [categories, setCategories] = useState<string[]>(['Common misconception', 'Form similar to answer'])
```
Comment (jhw123) [inline src/Components/CategoryInput.tsx:15]: These values are dummy values currently and will be fetched from the server, right?

**Review by jhw123 — APPROVED** [review]

File: src/Components/CategoryInput.tsx (hunk 1 above)
Comment (soyeongmin) [reply src/Components/CategoryInput.tsx:31]: okay.! Actually I don't understand the `useCallback` fully, so I should study it more.

File: src/Components/CategoryInput.tsx (hunk 3 above)
Comment (soyeongmin) [reply src/Components/CategoryInput.tsx:15]: Yep the categories will be replaced, but the inputCat is not.

**Pull request merged**